src/engine/controls/bpmcontrol.cpp

```diff
@@ -184,26 +190,15 @@ void BpmControl::slotAdjustBeatsSlower(double v) {
 }
 
 void BpmControl::slotTranslateBeatsEarlier(double v) {
-    if (v <= 0) {
-        return;
-    }
-    const TrackPointer pTrack = getEngineBuffer()->getLoadedTrack();
-    if (!pTrack) {
-        return;
-    }
-    const mixxx::BeatsPointer pBeats = pTrack->getBeats();
-    if (pBeats) {
-        const double sampleOffset = frameInfo().sampleRate * -0.01;
-        const mixxx::audio::FrameDiff_t frameOffset = sampleOffset / mixxx::kEngineChannelCount;
-        const auto translatedBeats = pBeats->tryTranslate(frameOffset);
-        if (translatedBeats) {
-            pTrack->trySetBeats(*translatedBeats);
-        }
-    }
+    slotTranslateBeatsMove(v);
 }
 
 void BpmControl::slotTranslateBeatsLater(double v) {
-    if (v <= 0) {
+    slotTranslateBeatsMove(v);
+}
+
+void BpmControl::slotTranslateBeatsMove(double v) {
+    if (v == 0) {
```

Comment:
I think we need to cast to integer here so the behaviour still matches the description. I.e. round fractional values so we get a constant step size.
```suggestion
    v = std::round(v);
    if (v == 0) {
```